otj-server-reactive/src/test/java/com/opentable/server/reactive/TestReactiveServerConfiguration.java

```diff
@@ -55,15 +54,32 @@ WebClient webClient() {
         return WebClient.builder().build();
     }
 
+    /**
+     * This scheduler is used by {@link ConservedHeadersWebFilterTest}
+     * to set up fewer available threads than the number of requests we want to make in parallel, to ensure that
+     * reactive operations from different requests are scheduled on threads that were previously used by another
+     * request.
+     */
+    @Bean
+    Scheduler getScheduler() {
+        return Schedulers.newParallel("parallel-scheduler", 3);
+    }
+
     @RestController
     @RequestMapping("/api")
     public static class MyResource {
 
+        private static final Logger LOG = LoggerFactory.getLogger(MyResource.class);
+
         @Autowired
         private WebClient webClient;
 
+        @Autowired
+        private Scheduler parallelScheduler;
+
         @RequestMapping("test")
         public Mono<String> test() {
+            LOG.info("Called 'test' endpoint");
             return Mono.just("test");
         }
 
```

Comment:
btw I just noticed echo - isn't that an outside service this test depends on.

Reply:
As discussed, we need to be able to call out to some sort of external service. This is already done in otj-server-mvc. We could consider refactoring and testing in a different way (WireMock/MockServer?) for both modules.